docs/CIP/CIP-XXXX/README.md

```diff
@@ -0,0 +1,268 @@
+
+### Message Types & Structure
+
+We distinguish between client-sent and server-sent messages.
+
+#### Client
+
+Client-sent messages are **typed** messages, such as:
+
+```json
+{
+  "type": "<unique message type>"
+  // ...
+}
+```
+
+This allows the server to quickly identify the client's intent. More details on what typed messages exist will be covered in [Client Typed Messages](#client-typed-messages) section.
+
+#### Server
+
+Instead of typed messages, server-sent messages vary in structure depending on the underlying blockchain. Our design aims to be chain-agnostic and serve data from different blockchains in future. Therefore, we rely on **untyped** messages that are enriched and aggregated into compound messages. We call each top level key a server message partial.
+
+This leads to a structure that has many top level keys of which, not every key may be present for every message, for example:
+
+```json
+{
+  "transactions": [
+    /* ... */
+  ],
+  "resolvedInputs": [
+    /* ... */
+  ],
+  "rewards": {
+    /* ... */
+  },
+  "tip": {
+    /* ... */
+  }
+  // ...
+}
+```
+
+> [!IMPORTANT]
+>
+> New message partials can be added at any time, which allows for flexible extension of the API.
+> Clients are expected to check for message key presence and ignore those they do not support.
+
+#### Errors
+
+Any cases of server-side failure trigger an error message that is append as separate `error` message partial:
+
+```json
+{
+  "error": {
+    "type": "error type",
+    "message": "error message"
+  }
+  /* ... */
+}
+```
+
+### Authentication
+
+A newly connected client authenticates as part of the [`subscribe`](./messages/client/subscribe.md) message that is sent to the server to synchronize a specific wallet account up to the chain's current tip. This message must include a **signature** that verifies the ownership of provided extened public key (xpub key).
+
+Extended public keys (xpub keys) serve a dual purpose in our design: **authentication** and **efficient transaction querying**. Here's how it works:
+
+#### Account Authentication
+
+Clients connect and authenticate via subscribe messages by providing:
+
+- a digital signature
+- their extended account public key (xpub key)
+
+This ensures that only authorized clients can access their account data.
+
+The `signature` field is generated by signing with the [extended account private key](https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf) a SHA256 HMAC hashed string. The prehash string is constructed by concatenating the subscribed blockchain(s) + timestamp.
+
+The server verifies the signature and timestamp before serving any data to the client. If any part of the `subscribe` message is invalid, the server sends an error and closes the connection.
```

Comment:
Should we mention a timestamp window since we will use `timestamp` for signature verification and try to guess it on the server? Should the timestamp be in a 60ms window, meaning the server tries 60 times before failing?